W-18518517 A2A Connector (#2906)
mulesoft/docs-connectors: a2a/1.0/modules/ROOT/pages/a2a-connector-reference.adoc

= A2A Connector 0.1.0-BETA Reference

Anypoint Connector for Agent2Agent (A2A) (A2A Connector) provides A2A protocol support for LLM agents, enabling your agents to talk to other agents.

[NOTE]
====

> Anypoint Connector for Agentforce (A2A) is a beta connector that is subject to the Beta Services Terms at https://www.salesforce.com/company/legal/agreements/[Agreements - Salesforce.com] or a written Unified Pilot Agreement if executed by Customer, and the Non-GA Gen AI and Non-GA Credit Consumption terms in the https://ptd.salesforce.com/?_ga=2.247987783.1372150065.1709219475-629000709.1639001992[Product Terms Directory]. Use of this beta connector with external systems or services consumes https://docs.mulesoft.com/general/usage-metrics[Mule messages] and is at the Customer's sole discretion. 

Reviewer: ```suggestion
Anypoint Connector for Agent2Agent (A2A) is a beta connector that is subject to the Beta Services Terms at https://www.salesforce.com/company/legal/agreements/[Agreements - Salesforce.com] or a written Unified Pilot Agreement if executed by Customer, and the Non-GA Gen AI and Non-GA Credit Consumption terms in the https://ptd.salesforce.com/?_ga=2.247987783.1372150065.1709219475-629000709.1639001992[Product Terms Directory]. Use of this beta connector with external systems or services consumes https://docs.mulesoft.com/general/usage-metrics[Mule messages] and is at the Customer's sole discretion. 
```

> Anypoint Connector for Agentforce (A2A) is a beta connector that is subject to the Beta Services Terms at https://www.salesforce.com/company/legal/agreements/[Agreements - Salesforce.com] or a written Unified Pilot Agreement if executed by Customer, and the Non-GA Gen AI and Non-GA Credit Consumption terms in the https://ptd.salesforce.com/?_ga=2.247987783.1372150065.1709219475-629000709.1639001992[Product Terms Directory]. Use of this beta connector with external systems or services consumes https://docs.mulesoft.com/general/usage-metrics[Mule messages] and is at the Customer's sole discretion. 

Reviewer: ```suggestion
Anypoint Connector for Agentforce (A2A) is a beta connector that is subject to the Beta Services Terms at https://www.salesforce.com/company/legal/agreements/[Agreements - Salesforce.com] or a written Unified Pilot Agreement if executed by Customer, and the Non-GA Gen AI and Non-GA Credit Consumption terms in the https://ptd.salesforce.com/?_ga=2.247987783.1372150065.1709219475-629000709.1639001992[Product Terms Directory]. Use of this beta connector with external systems or services consumes xref:general::usage-metrics.adoc[Mule messages] and is at the Customer's sole discretion. 
```

====

== Configurations

[[Client]]
=== Client

The application or AI system that initiates requests and consumes services offered by A2A servers. This allows Mule apps to act as A2A brokers between non A2A “head” agents and A2A “node” agents.

==== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
|Name | String | Name for this configuration. Connectors reference the configuration with this name. | | x

> | Connection a| <<Client_ClientConnection, Client Connection>>

Reviewer: ```suggestion
| Connection a| <<ClientConnection, Client Connection>>
```

 | The connection types to provide to this configuration. | | x
| Name a| String |  ID used to reference this configuration. |  | x
| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration.   |  | 
|===

==== Connection Types

[[ClientConnection]]
===== Client Connection

====== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Server Url a| String |  Base URL of the target A2A server. |  | x
| Request Timeout a| Number | Sets the duration to wait for server responses before timing out requests. |  30 | 
| Request Timeout Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS

> ** DAYS |  Time Unit for the requestTimeout parameter |  SECONDS | 

Reviewer: ```suggestion
** DAYS |  Time unit for the *Request Timeout* field. |  SECONDS | 
```

| Authentication a| HttpRequestAuthentication |  |  | 
| Tls Context a| <<Tls>> |  |  | 
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  | 
|===

---
[[Server]]
=== Server

The A2A server acts as a remote agent that exposes an HTTP(S) endpoint and implements the A2A protocol methods for interacting with other AI agents. This allows Mule-based agents to receive and respond to A2A protocol-compliant requests.

==== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
|Name | String | Name for this configuration. Connectors reference the configuration with this name. | | x

> | Connection a| <<Server_Connection, Connection>> | The connection types to provide to this configuration. | | x

Reviewer: ```suggestion
| Connection a| <<ServerConnection, Server Connection>> | The connection types to provide to this configuration. | | x
```

Reviewer: double check this one

|===
| Field | Type | Description | Default Value | Required
| Path a| String | Path to the CRL file. |  | 
|===

=== Reconnection

Configures a reconnection strategy for an operation.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Fails Deployment a| Boolean | When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  | 
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> | Reconnection strategy to use. |  | 
|===

=== Reconnect

Configures a standard reconnection strategy, which specifies how often to reconnect and how many reconnection attempts the connector source or operation can make.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Frequency a| Number | How often to attempt to reconnect, in milliseconds. |  | 
| Blocking a| Boolean | If `false`, the reconnection strategy runs in a separate, non-blocking thread. | true  | 
| Count a| Number | How many reconnection attempts the Mule app can make. |  | 
|===

> 

Reviewer: add [[ReconnectForever]]

|===

[[AgentCapabilities]]
=== Agent Capabilities

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Streaming a| Boolean | Set to `true` to support SSE streaming methods. | false | 
| Push Notifications a| Boolean | Set to `true` to support push notification methods. | false | 
| State Transition History a| Boolean |  | false | 
|===

[[AgentProvider]]
=== Agent Provider

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Organization a| String | Name of the organization or entity. |  | x
| Url a| String | URL of the provider's website. |  | 
|===

[[AgentAuthentication]]
=== Agent Authentication

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Authentication a| Array of String |  |  | x
| Credentials a| String |  |  | 
|===

[[RequestAttributes]]
=== Request Attributes

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Query Params a| Object |  |  | x
| Headers a| Object | Map of HTTP headers in the message. |  | x
|===

> 

Reviewer: add [[RedeliveryPolicy]]

=== Redelivery Policy

Configures the redelivery policy for executing requests that generate errors. You can add a redelivery policy to any source in a flow.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Max Redelivery Count a| Number | Maximum number of times to process a redelivered request unsuccessfully before returning a REDELIVERY_EXHAUSTED error. |  | 
| Message Digest Algorithm a| String | Secure hashing algorithm to use if the *Use Secure Hash* field is `true`. If the payload of the message is a Java object, Mule ignores this value and returns the value that the payload's `hashCode()` returned. |  | 
| Message Identifier a| <<RedeliveryPolicyMessageIdentifier>> | One or more expressions that determine if a message was redelivered. This property can be set only if the *Use Secure Hash* field is `false`. |  | 
| Object Store a| ObjectStore | Configures the object store that stores the redelivery counter for each message. |  | 
|===

> 

Reviewer: add [[RedeliveryPolicyMessageIdentifier]]
